### PR TITLE
OS Activate Target

### LINKED ISSUE #113: Implement OS Activate RPC for mock target

### PR DESCRIPTION
**Pull request by Jared-Mullin**
Related issue: Closes #113 
Added OS Activate service and tests

### CODE REVIEW

File: gnoi/os/server_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+			},
+			want: &pb.ActivateResponse{Response: &pb.ActivateResponse_ActivateError{
+				ActivateError: &pb.ActivateError{Type: 1},
+			}},
+		},
+	}
+	for _, test := range tests {
+		got, _ := server.Activate(context.Background(), test.request)
+		diff := reflect.DeepEqual(test.want, got)
+		if !diff {
```
Comment (loriscn12) [inline gnoi/os/server_test.go:43]: use either cmp or pretty/messagediff, more context: refelct.DeepEqual it is sesntitive to changes in unexported fields.

File: gnoi/os/server.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+// Server is an OS Management service.
+type Server struct {
+	pb.OSServer
+	manager *Manager
+}
+
+// NewServer returns an OS Management service.
+func NewServer(factoryVersion string) *Server {
+	return &Server{manager: NewManager(factoryVersion)}
```
Comment (samribeiro) [inline gnoi/os/server.go:33]: I think it is worth using a OSSettings struct defined in manager.go to pass:
factoryVersion string
installedVersions []string

just like reset does.

File: gnoi/os/server.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+// Register registers the server into the the gRPC server provided.
+func (s *Server) Register(g *grpc.Server) {
+	pb.RegisterOSServer(g, s)
+}
+
+// Activate sets the requested OS version as the version which is used at the next reboot, and reboots the Target.
+func (s *Server) Activate(ctx context.Context, request *pb.ActivateRequest) (*pb.ActivateResponse, error) {
+	if err := s.manager.SetRunning(request.Version); err != nil {
+		return &pb.ActivateResponse{Response: &pb.ActivateResponse_ActivateError{
+			ActivateError: &pb.ActivateError{Type: 1},
```
Comment (samribeiro) [inline gnoi/os/server.go:45]: use the generated code to make it readable:
https://github.com/google/gnxi/blob/master/gnoi/os/pb/os.pb.go#L121

File: gnoi/os/server.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+func NewServer(factoryVersion string) *Server {
+	return &Server{manager: NewManager(factoryVersion)}
+}
+
+// Register registers the server into the the gRPC server provided.
+func (s *Server) Register(g *grpc.Server) {
+	pb.RegisterOSServer(g, s)
+}
+
+// Activate sets the requested OS version as the version which is used at the next reboot, and reboots the Target.
```
Comment (samribeiro) [inline gnoi/os/server.go:41]: It is probably worth logging the server state change in the server.

File: gnoi/os/server_test.go (hunk 1 above)
Comment (Jared-Mullin) [reply gnoi/os/server_test.go:43]: Thanks! Will change it

File: gnoi/os/server_test.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+				Version: "99.0a",
+			},
+			want: &pb.ActivateResponse{Response: &pb.ActivateResponse_ActivateError{
+				ActivateError: &pb.ActivateError{Type: pb.ActivateError_NON_EXISTENT_VERSION},
+			}},
+		},
+	}
+	for _, test := range tests {
+		got, _ := server.Activate(context.Background(), test.request)
+		diff := pretty.Compare(test.want.Response, got.Response)
```
Comment (loriscn12) [inline gnoi/os/server_test.go:60]: if diff := pretty.Compare(test.want.Response, got.Response); diff != "" {

**Pull request merged**